From a7224619eb412498436525f21c8801d7f201e1f6 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Mon, 4 Dec 2023 21:32:09 +0100
Subject: [PATCH] bluez5: Set Modalias, Class & Icon properties

Otherwise we only emit those with PropertiesChanged but GetAll won't
return those properties.
---
 dbusmock/templates/bluez5.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dbusmock/templates/bluez5.py b/dbusmock/templates/bluez5.py
index c94946f..dd3bf91 100644
--- a/dbusmock/templates/bluez5.py
+++ b/dbusmock/templates/bluez5.py
@@ -474,6 +474,10 @@ def PairDevice(_self, adapter_device_name, device_address, class_):
             'Icon': dbus.String('phone', variant_level=1),
         })
 
+    device.props[DEVICE_IFACE]['Modalias'] = dbus.String("bluetooth:v000Fp1200d1436", variant_level=1)
+    device.props[DEVICE_IFACE]['Class'] = dbus.UInt32(class_, variant_level=1)
+    device.props[DEVICE_IFACE]['Icon'] = dbus.String("phone", variant_level=1)
+
     device.EmitSignal(dbus.PROPERTIES_IFACE, 'PropertiesChanged', 'sa{sv}as', [
         DEVICE_IFACE,
         {
-- 
2.43.0

